src/CMakeLists.txt

```diff
@@ -229,11 +230,19 @@ if("${CMAKE_C_SIMULATE_ID}" STREQUAL "MSVC")
                          PRIVATE
                            -Xclang -fblocks)
 else()
-  # FIXME(compnerd) add check for -momit-leaf-frame-pointer?
-  target_compile_options(dispatch
-                         PRIVATE
-                           -fblocks
-                           -momit-leaf-frame-pointer)
+  check_c_compiler_flag("-momit-leaf-frame-pointer -Werror -Wall -O3" C_SUPPORTS_OMIT_LEAF_FRAME_POINTER)
```

Comment:
`-Werror -Wall -O3` is not required when using `check_c_compiler_flag`.  See the documentation at https://cmake.org/cmake/help/latest/module/CheckCCompilerFlag.html.

I think that we should just do two `target_compile_options` calls here.

```
check_c_compile_flag("-momit-leaf-frame-pointer" CC_SUPPORTS_MOMIT_LEAF_FRAME_POINTER)
target_compile_options(dispatch PRIVATE -fblocks)
if(CC_SUPPORTS_MOMIT_LEAF_FRAME_POINTER)
  target_compile_options(dispatch PRIVATE -momit-leaf-frame-pointer)
endif()

```

Reply:
check_c_compiler_flag just runs the compilation command with the provided flag. Unfortunately, clang will only complain with this combination of flags.

Reply:
Blah, that is annoying.